src/cmap/connection.ts

```diff
@@ -427,11 +429,8 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
   ): AsyncGenerator<MongoDBResponse> {
     this.throwIfAborted();
 
-    if (typeof options.socketTimeoutMS === 'number') {
-      this.socket.setTimeout(options.socketTimeoutMS);
-    } else if (this.socketTimeoutMS !== 0) {
-      this.socket.setTimeout(this.socketTimeoutMS);
-    }
+    const socketTimeoutMS = options.socketTimeoutMS ?? this.socketTimeoutMS ?? 0;
+    this.socket.setTimeout(socketTimeoutMS);
```

Comment:
Could this change be done separately from this bug fix or does it relate directly? IIRC the "not zero" condition was actually significant rather than an elimination of null (where as the typeof number was probably just eliminating null, but I do question removing the type safety)

Reply:
Yeah, it can be done separately.  I can revert this if you'd like, just let me know.  But to your point - it makes no difference:

- when not in use, socket timeouts are set to 0.
- if the options socket timeout is non-null, we should use that.
- if the socket timeout on the connection is non-null, we should use that.  if it is explicitly zero, then this.socket.setTimeout(0) is no change in behavior from _not_ setting it, because it's already defaulted to 0.

Reply:
I think you are correct, I just remember writing this exact improvement and having to revert it back to these conditions when I refactored the connection layer. Digging through history now, I'll let you know if I come up empty.

Reply:
FYI sent you some historical context on slack. Let me know if that changes things here. But I think we can keep this once we get our CI back on track and nothing fails.

Reply:
Given the unreliability of our CI right now, I reverted the changes.